source/blog/2018-05-30-emberjs-accessibility-and-a11y-tools.md

```diff
@@ -0,0 +1,153 @@
+```
+
+![A form with labels](/images/blog/2018-05-30-emberjs-accessibility-and-a11y-tools/form-with-labels.png)
+
+Now the filter has a label, and the tests pass!
+
+### Using ember-a11y-landmarks for roles
+
+The [ember-a11y-landmarks](https://github.com/ember-a11y/ember-a11y-landmarks) addon helps manage the roles that should go on semantic tags like `<header>` and `<nav>`, without needing to learn the intricacies of when to use them.
+
+One huge thing you can do to improve an app's accessibility is to use semantic tags in the HTML. For example, always use `<button>` for buttons, instead of making them `divs`. However, some semantic tags still need extra attributes called roles for screen readers to work right, and they only need them some of the time based on their position in the DOM. ember-a11y-landmarks to the rescue!
+
+To use this addon, wrap content in the ally-landmark block and say what kind of tag to use. Here's a template example:
+
+```handlebars
+{{#a11y-landmark tagName="footer"}}
+   This is a footer!
+{{/a11y-landmark}}
+And here's the html output with the correct role assigned:
+<footer id="ember337" role="contentinfo" class="ember-view"> 
+  This is a footer!
+</footer>
+```
+
+## Structural application concerns
+
+I learned from [Robert DeLuca](https://github.com/Robdel12) that [most front end frameworks have flaws](https://medium.com/@robdel12/single-page-apps-routers-are-broken-255daa310cf) that affect Assistive Technology like screen readers. However, there's a lot of progress being made in Ember itself, our community addons, and by our pals working in other frameworks! You can learn more in the recent JavaScript [State of Accessibility](https://www.youtube.com/watch?v=aoyLG2gTFpI) panel hosted by [ThisDot Media](https://www.thisdot.co/). Jump to the Ember segment presented by Melanie Sumner [here](https://youtu.be/aoyLG2gTFpI?t=2432).
+
+With the release of 3.1, Ember.js introduced some new features within the framework itself, and community addons continue to help drive progress on the remaining concerns. Historically, many new features in Ember originated as community addons.
+
+### New options for out-of-the-box accessibility
+
+One common issue in single page application frameworks is that they wrap a developer's content in extra `divs`. However, some elements like `<header>` should be direct descendants of the `body` element. [RFC 280](https://github.com/emberjs/rfcs/blob/master/text/0280-remove-application-wrapper.md) proposed an option for removing the application-level `div` wrapper (an RFC, or Request for Comments, is the main way that new features are proposed and opened for community feedback). That RFC was accepted, meaning that work could begin to add the feature into Ember, and the new option shipped in version 3.1. You can learn more about it and how to enable the feature in the [release blog post](https://www.emberjs.com/blog/2018/04/13/ember-3-1-released.html#toc_introducing-optional-features-3-of-4).
```

Comment:
This section removes one of the use cases from ember-a11y-landmarks. I think that this paragraph should be worked into the previous section, "Using ember-a11y-landmarks for roles".

Reply:
I hint at it now in that section. I would like to maintain the divide between addons and "officially part of Ember" to help tell the story of how addon features can be sometimes added as base features of Ember